pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -577,28 +604,136 @@ func (r *SubnetPortReconciler) addressBindingMapFunc(ctx context.Context, obj cl
+}
+
+func setAddressBindingStatusBySubnetPort(client client.Client, ctx context.Context, subnetPort *v1alpha1.SubnetPort, subnetPortService *subnetport.SubnetPortService, transitionTime metav1.Time, e error) {
+	ipAddress := ""
 	subnetPortID := subnetPortService.BuildSubnetPortId(&subnetPort.ObjectMeta)
 	nsxSubnetPort := subnetPortService.SubnetPortStore.GetByKey(subnetPortID)
 	if nsxSubnetPort == nil {
 		log.Info("Missing SubnetPort", "id", subnetPort.UID)
-		return
-	}
-	if nsxSubnetPort.ExternalAddressBinding == nil || nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress == nil {
-		return
+		if e == nil {
+			e = vmOrInterfaceNotFoundError
+		}
+	} else if nsxSubnetPort.ExternalAddressBinding != nil && nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress != nil {
+		ipAddress = *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress
 	}
 	ab := subnetPortService.GetAddressBindingBySubnetPort(subnetPort)
 	if ab == nil {
-		log.Info("Missing AddressBinding for SubnetPort", "namespace", subnetPort.Namespace, "name", subnetPort.Name)
+		log.Info("No AddressBinding for SubnetPort", "namespace", subnetPort.Namespace, "name", subnetPort.Name)
 		return
 	}
-	if ab.Status.IPAddress != *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress {
+	setAddressBindingStatus(client, ctx, ab, transitionTime, e, ipAddress)
+}
+
+func setAddressBindingStatus(client client.Client, ctx context.Context, ab *v1alpha1.AddressBinding, transitionTime metav1.Time, e error, ipAddress string) {
+	newConditions := newReadyCondition("AddressBinding", transitionTime, e)
+	isUpdated := false
+	for i := range newConditions {
+		conditionUpdated := false
+		ab.Status.Conditions, conditionUpdated = mergeCondition(ab.Status.Conditions, &newConditions[i])
+		isUpdated = isUpdated || conditionUpdated
+	}
+	if ab.Status.IPAddress != ipAddress {
+		isUpdated = true
+	}
+	if isUpdated {
 		ab = ab.DeepCopy()
-		ab.Status.IPAddress = *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress
+		ab.Status.IPAddress = ipAddress
 		client.Status().Update(ctx, ab)
 		log.V(1).Info("Updated AddressBinding CR status", "namespace", ab.Namespace, "name", ab.Name, "status", ab.Status)
 	}
 }
+
+func newReadyCondition(resourceType string, transitionTime metav1.Time, e error) []v1alpha1.Condition {
+	if e == nil {
+		return []v1alpha1.Condition{
+			{
+				Type:               v1alpha1.Ready,
+				Status:             v1.ConditionTrue,
+				Message:            fmt.Sprintf("%s has been successfully created/updated", resourceType),
+				Reason:             fmt.Sprintf("%sReady", resourceType),
+				LastTransitionTime: transitionTime,
+			},
+		}
+	}
+	return []v1alpha1.Condition{
+		{
+			Type:               v1alpha1.Ready,
+			Status:             v1.ConditionFalse,
+			Message:            fmt.Sprintf("error occurred while processing the %s CR. Error: %v", resourceType, e),
+			Reason:             fmt.Sprintf("%sNotReady", resourceType),
+			LastTransitionTime: transitionTime,
+		},
+	}
+}
```

Comment:
```suggestion
func newReadyCondition(resourceType string, transitionTime metav1.Time, err error) []v1alpha1.Condition {
	condition := v1alpha1.Condition{
		Type:               v1alpha1.Ready,
		LastTransitionTime: transitionTime,
	}

	if err == nil {
		condition.Status = v1.ConditionTrue
		condition.Message = fmt.Sprintf("%s has been successfully created/updated", resourceType)
		condition.Reason = fmt.Sprintf("%sReady", resourceType)
	} else {
		condition.Status = v1.ConditionFalse
		condition.Message = fmt.Sprintf("error occurred while processing the %s CR. Error: %v", resourceType, err)
		condition.Reason = fmt.Sprintf("%sNotReady", resourceType)
	}

	return []v1alpha1.Condition{condition}
}
```

Reply:
There is no much difference for this style change. We can keep existing one.

Reply:
The current implementation is OK, just small changes and not essential. The difference between the two condition can be clearly seen.